src/events/tls.py

```diff
@@ -296,6 +297,13 @@ def _request_certificate(self):
 
         sans = self.charm.broker.tls_manager.build_sans()
 
+        # only warn during certificate creation, not every event if in structured_config
+        if self.charm.config.certificate_extra_sans:
+            warnings.warn(
+                "'certificate_extra_sans' config option is deprecated, use 'extra_listeners' instead",
+                DeprecationWarning,
+            )
+
```

Comment:
**suggestion**: Could we add a deprecation notice to the config.yaml file as well? Feel free to disregard if you don't want to re-run everything for such a small change?